src/openrct2-ui/windows/ScenarioSelect.cpp

```diff
@@ -254,433 +207,414 @@ static void WindowScenarioselectInitTabs(WindowBase* w)
+        _listItems.clear();
+        _listItems.shrink_to_fit();
+    }
 
-        ShortenPath(path, sizeof(path), scenario->Path, w->width - 6 - TabWidth, FontStyle::Medium);
+    void OnPrepareDraw() override
+    {
+        pressed_widgets &= ~(
+            (1uLL << WIDX_CLOSE) | (1uLL << WIDX_TAB1) | (1uLL << WIDX_TAB2) | (1uLL << WIDX_TAB3) | (1uLL << WIDX_TAB4)
+            | (1uLL << WIDX_TAB5) | (1uLL << WIDX_TAB6) | (1uLL << WIDX_TAB7) | (1uLL << WIDX_TAB8) | (1uLL << WIDX_TAB9)
+            | (1uLL << WIDX_TAB10));
 
-        const utf8* pathPtr = path;
-        auto ft = Formatter();
-        ft.Add<const char*>(pathPtr);
-        DrawTextBasic(
-            *dpi, w->windowPos + ScreenCoordsXY{ TabWidth + 3, w->height - 3 - 11 }, STR_STRING, ft, { w->colours[1] });
+        pressed_widgets |= 1LL << (selected_tab + WIDX_TAB1);
+
+        ResizeFrameWithPage();
+        const int32_t bottomMargin = gConfigGeneral.DebuggingTools ? 17 : 5;
+        window_scenarioselect_widgets[WIDX_SCENARIOLIST].right = width - 179;
+        window_scenarioselect_widgets[WIDX_SCENARIOLIST].bottom = height - bottomMargin;
     }
 
-    // Scenario name
-    auto screenPos = w->windowPos
-        + ScreenCoordsXY{ window_scenarioselect_widgets[WIDX_SCENARIOLIST].right + 4,
-                          window_scenarioselect_widgets[WIDX_TABCONTENT].top + 5 };
-    auto ft = Formatter();
-    ft.Add<StringId>(STR_STRING);
-    ft.Add<const char*>(scenario->Name);
-    DrawTextEllipsised(
-        *dpi, screenPos + ScreenCoordsXY{ 85, 0 }, 170, STR_WINDOW_COLOUR_2_STRINGID, ft, { TextAlignment::CENTRE });
-    screenPos.y += 15;
-
-    // Scenario details
-    ft = Formatter();
-    ft.Add<StringId>(STR_STRING);
-    ft.Add<const char*>(scenario->Details);
-    screenPos.y += DrawTextWrapped(*dpi, screenPos, 170, STR_BLACK_STRING, ft) + 5;
-
-    // Scenario objective
-    ft = Formatter();
-    ft.Add<StringId>(ObjectiveNames[scenario->ObjectiveType]);
-    if (scenario->ObjectiveType == OBJECTIVE_BUILD_THE_BEST)
+    ScreenSize OnScrollGetSize(int32_t scrollIndex) override
     {
-        StringId rideTypeString = STR_NONE;
-        auto rideTypeId = scenario->ObjectiveArg3;
-        if (rideTypeId != RIDE_TYPE_NULL && rideTypeId < RIDE_TYPE_COUNT)
+        const int32_t scenarioItemHeight = GetScenarioListItemSize();
+
+        int32_t y = 0;
+        for (const auto& listItem : _listItems)
         {
-            rideTypeString = GetRideTypeDescriptor(rideTypeId).Naming.Name;
+            switch (listItem.type)
+            {
+                case ListItemType::Heading:
+                    y += 18;
+                    break;
+                case ListItemType::Scenario:
+                    y += scenarioItemHeight;
+                    break;
+            }
         }
-        ft.Add<StringId>(rideTypeString);
+
+        return { WW, y };
     }
-    else
+    void OnScrollMouseOver(int32_t scrollIndex, const ScreenCoordsXY& screenCoords) override
```

Comment:
Empty line between functions
```suggestion

    void OnScrollMouseOver(int32_t scrollIndex, const ScreenCoordsXY& screenCoords) override
```